scripts/periodic_cleanup.py

```diff
@@ -46,6 +53,16 @@ def cleanup_instances(cs_client, age_limit, instance_prefix):
             server.delete()
 
 
+def cleanup_jenkins_nodes(jenkins_client, instance_prefix):
+    for node_id, node in jenkins_client.get_nodes().iteritems():
+        # Ignore nodes that are coming online for the first time
+        # which won't have an offline cause.
+        if (node_id.startswith(instance_prefix) and not node.is_online()
+                and node._data.get("offlineCauseReason")):
```

Comment:
Reaching into the class for `_data` is not ideal. I think it would be better to use an alternative if possible. I had a skim through the module and it looks like `node.poll().get("offlineCauseReason")` should work.